### ISSUE TITLE
Filter type doesn't update when column changes, causing strange behaviour

### ISSUE THREAD

**Issue by alexpeattie** [trey-wallis/obsidian-dataloom#635]
### Describe the bug

Currently if I change the column a filter is targeting, the filter type is inherited from the previous state, even if that's incompatible with the new column. For example, if I have a filter on a text column with type `is` and edit it to target a different column type, the `type` will remain `is` - even if the `is` filter type isn't compatible with new column type.

### Steps to reproduce

The easiest way to reproduce the bug is to switch column of a filter to a multi-tag column, and not touch the type (which will visually show "Contain" but whose value will be `is` under the hood). See the video below:

https://github.com/trey-wallis/obsidian-dataloom/assets/636814/5df8881f-397c-40a5-9f02-60fc1027398f

(In the console, I'm just logging the arguments of `doesTagMatch`)

### Expected behavior

The underlying filter type should match what's shown in the UI.

My instinct is that in `handleRuleColumnChange` we might also need to change the filter type (to an arbitrary compatible filter type for the newly selected column's type), but not sure if this is the best solution 🙂 .

P.S. Thanks for your work on this great plugin 🙏 

### Are you using the mobile app?

No

### Obsidian debug info

```
SYSTEM INFO:
	Obsidian version: v1.3.5
	Installer version: v1.3.5
	Operating system: Darwin Kernel Version 20.6.0: Wed Jun 23 00:26:27 PDT 2021; root:xnu-7195.141.2~5/RELEASE_ARM64_T8101 20.6.0
	Login status: not logged in
	Insider build toggle: off
	Live preview: on
	Legacy editor: off
	Base theme: light
	Community theme: none
	Snippets enabled: 1
	Restricted mode: off
	Plugins installed: 19
	Plugins enabled: 14
		[Redacting for privacy]
		14: DataLoom v8.0.0
```

**Comment by trey-wallis** [comment]
Hi @alexpeattie. Thanks for finding this. This may be related to #662. I will look into it this week

**Comment by trey-wallis** [comment]
Fixed in 8.5.0 :)
